crates/compiler/can/src/def.rs

```diff
@@ -2880,6 +2941,188 @@ fn to_pending_value_def<'a>(
             preceding_comment: *preceding_comment,
         }),
 
+        ModuleImport(module_import) => {
+            let module_name = ModuleName::from(module_import.name.value.name.as_str());
+            let pq_module_name = QualifiedModuleName {
+                opt_package: module_import.name.value.package,
+                module: module_name.clone(),
+            }
+            .into_pq_module_name(env.opt_shorthand);
+
+            let module_id = env
+                .qualified_module_ids
+                .get_id(&pq_module_name)
+                .expect("Module id should have been added in load");
+
+            let name_with_alias = match module_import.alias {
+                Some(alias) => ModuleName::from(alias.item.value.as_str()),
+                None => module_name.clone(),
+            };
+
+            if let Err(existing_import) =
+                scope
+                    .modules
+                    .insert(name_with_alias.clone(), module_id, region)
+            {
+                env.problems.push(Problem::ImportNameConflict {
+                    name: name_with_alias,
+                    is_alias: module_import.alias.is_some(),
+                    new_module_id: module_id,
+                    new_import_region: region,
+                    existing_import,
+                });
+
+                return PendingValue::ImportNameConflict;
+            }
+
+            let mut exposed_symbols;
+
+            let is_automatically_imported =
+                !env.home.is_builtin() && module_id.is_automatically_imported();
+
+            match module_import.exposed {
```

Comment:
I think this might be a bit easier to read with less branches. What do you think of something like

```
let exposed_ids = env.dep_idents.get(&module_id).unwrap_or_default();

if exposed_ids.is_empty() { /* error about explicit builtin import */ }

for exposed_id in exposed_ids {
  // existing logic in Some branch below
}
```

I think this also has the advantage that it will report a problem for an import like `Num exposing []`